File: Year 1/Semester 1/Java/Lessons/enumerations.java
public class enumerations {
    // Define an enum
public enum Day {
    MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY
}

// Using the enum in a class
public class EnumExample {
    
    // Method to demonstrate the use of enums
    public void printDayMessage(Day day) {
        switch (day) {
            case MONDAY:
                System.out.println("Start of the work week!");
                break;
            case TUESDAY:
                System.out.println("Second day of the work week.");
                break;
            case WEDNESDAY:
                System.out.println("Midweek day.");
                break;
            case THURSDAY:
                System.out.println("Almost the weekend.");
                break;
            case FRIDAY:
                System.out.println("Last work day of the week!");
                break;
            case SATURDAY:
                System.out.println("Weekend fun day!");
                break;
            case SUNDAY:
                System.out.println("Rest and relaxation day.");
                break;
        }
    }

    public static void main(String[] args) {
        // Creating an instance of EnumExample
        EnumExample example = new EnumExample();

        // Demonstrating the use of enums
        example.printDayMessage(Day.MONDAY);
        example.printDayMessage(Day.FRIDAY);
        example.printDayMessage(Day.SUNDAY);
    }
}

}

File: Year 1/Semester 1/C/Lessons/enumerations.c
#include <stdio.h>

// Define an enum for days of the week
typedef enum {
    MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY
} Day;

// Function to demonstrate the use of enums
void printDayMessage(Day day) {
    switch (day) {
        case MONDAY:
            printf("Start of the work week!\n");
            break;
        case TUESDAY:
            printf("Second day of the work week.\n");
            break;
        case WEDNESDAY:
            printf("Midweek day.\n");
            break;
        case THURSDAY:
            printf("Almost the weekend.\n");
            break;
        case FRIDAY:
            printf("Last work day of the week!\n");
            break;
        case SATURDAY:
            printf("Weekend fun day!\n");
            break;
        case SUNDAY:
            printf("Rest and relaxation day.\n");
            break;
    }
}

int main() {
    // Demonstrating the use of enums
    printDayMessage(MONDAY);
    printDayMessage(FRIDAY);
    printDayMessage(SUNDAY);

    return 0;
}
